OpenRA.Game/Network/Order.cs

```diff
@@ -127,17 +128,15 @@ public static Order Deserialize(World world, BinaryReader r)
 
 								case TargetType.Terrain:
 									{
-										if (world != null)
-											target = Target.FromCell(world, (CPos)r.ReadInt2());
-
+										target = Target.FromPos(new WPos(r.ReadInt32(), r.ReadInt32(), r.ReadInt32()));
```

Comment:
Terrain based targets used the `WPos pos` field for position - not the cell. See further down.